ofrak_core/ofrak/gui/server.py

```diff
@@ -300,63 +337,103 @@ async def get_resource_range(resource_id):
     @exceptions_to_http(SerializedError)
     async def unpack(self, request: Request) -> Response:
         resource = await self._get_resource_for_request(request)
-        result = await resource.unpack()
-        response_pjson = await self._serialize_component_result(result)
-        return json_response(response_pjson)
+        script_str = """
+        await {resource}.unpack()"""
+        await self.script_builder.add_action(resource, script_str, ActionType.UNPACK)
+        try:
+            result = await resource.unpack()
+            await self.script_builder.commit_to_script(resource)
+        except Exception as e:
+            await self.script_builder.clear_script_queue(resource)
+            raise e
+        return json_response(await self._serialize_component_result(result))
 
     @exceptions_to_http(SerializedError)
     async def unpack_recursively(self, request: Request) -> Response:
         resource = await self._get_resource_for_request(request)
-        result = await resource.unpack_recursively()
-        response_pjson = await self._serialize_component_result(result)
-        return json_response(response_pjson)
+        script_str = """
+        await {resource}.unpack_recursively()"""
+        await self.script_builder.add_action(resource, script_str, ActionType.UNPACK)
+        try:
+            result = await resource.unpack_recursively()
+            await self.script_builder.commit_to_script(resource)
+        except Exception as e:
+            await self.script_builder.clear_script_queue(resource)
+            raise e
+        return json_response(await self._serialize_component_result(result))
 
     @exceptions_to_http(SerializedError)
     async def pack(self, request: Request) -> Response:
         resource = await self._get_resource_for_request(request)
+        script_str = """
+
+        await {resource}.pack()"""
```

Comment:
```suggestion
        script_str = """
        await {resource}.pack()"""
```

This spacing is inconsistent with other component calls, but also with `pack_recursively` below. If the intention was to space out the `pack` call, then `pack_recursively` should be spaced out as well. Otherwise, this space should probably be removed.